pkg/snapstore/s3_snapstore.go

```diff
@@ -61,56 +73,57 @@ type S3SnapStore struct {
 func NewS3SnapStore(config *brtypes.SnapstoreConfig) (*S3SnapStore, error) {
-	credentials, err := getSessionOptions(getEnvPrefixString(config.IsSource))
+	credentials, sseCreds, err := getSessionOptions(getEnvPrefixString(config.IsSource))
 	if err != nil {
 		return nil, err
 	}
-	return newS3FromSessionOpt(config.Container, config.Prefix, config.TempDir, config.MaxParallelChunkUploads, config.MinChunkSize, credentials)
+	return newS3FromSessionOpt(config.Container, config.Prefix, config.TempDir, config.MaxParallelChunkUploads, config.MinChunkSize, credentials, sseCreds)
 }
 
 // newS3FromSessionOpt will create the new S3 snapstore object from S3 session options
-func newS3FromSessionOpt(bucket, prefix, tempDir string, maxParallelChunkUploads uint, minChunkSize int64, so session.Options) (*S3SnapStore, error) {
+func newS3FromSessionOpt(bucket, prefix, tempDir string, maxParallelChunkUploads uint, minChunkSize int64, so session.Options, sseCreds sseCredentials) (*S3SnapStore, error) {
 	sess, err := session.NewSessionWithOptions(so)
 	if err != nil {
 		return nil, fmt.Errorf("new AWS session failed: %v", err)
 	}
 	cli := s3.New(sess)
-	return NewS3FromClient(bucket, prefix, tempDir, maxParallelChunkUploads, minChunkSize, cli), nil
+	return NewS3FromClient(bucket, prefix, tempDir, maxParallelChunkUploads, minChunkSize, cli, sseCreds), nil
 }
 
-func getSessionOptions(prefixString string) (session.Options, error) {
+func getSessionOptions(prefixString string) (session.Options, sseCredentials, error) {
 
 	if filename, isSet := os.LookupEnv(prefixString + awsCredentialJSONFile); isSet {
-		creds, err := readAWSCredentialsJSONFile(filename)
+		creds, sseCreds, err := readAWSCredentialsJSONFile(filename)
 		if err != nil {
-			return session.Options{}, fmt.Errorf("error getting credentials using %v file", filename)
+			return session.Options{}, sseCredentials{}, fmt.Errorf("error getting credentials using %v file", filename)
 		}
-		return creds, nil
+		return creds, sseCreds, nil
 	}
 
 	if dir, isSet := os.LookupEnv(prefixString + awsCredentialDirectory); isSet {
-		creds, err := readAWSCredentialFiles(dir)
+		creds, sseCreds, err := readAWSCredentialFiles(dir)
 		if err != nil {
-			return session.Options{}, fmt.Errorf("error getting credentials from %v directory", dir)
+			return session.Options{}, sseCredentials{}, fmt.Errorf("error getting credentials from %v directory", dir)
 		}
-		return creds, nil
+		return creds, sseCreds, nil
 	}
 
 	return session.Options{
 		// Setting this is equal to the AWS_SDK_LOAD_CONFIG environment variable was set.
 		// We want to save the work to set AWS_SDK_LOAD_CONFIG=1 outside.
 		SharedConfigState: session.SharedConfigEnable,
-	}, nil
+	}, sseCredentials{}, nil
 }
 
-func readAWSCredentialsJSONFile(filename string) (session.Options, error) {
+func readAWSCredentialsJSONFile(filename string) (session.Options, sseCredentials, error) {
```

Comment:
Unfortunately there is a huge amount of code repetition in `readAWSCredentialsJSONFile` and `readAWSCredentialFiles` functions. Since these existed before i do not expect you to make these changes.